enterprise/ha/src/test/java/org/neo4j/kernel/ha/DelegateInvocationHandlerTest.java

```diff
@@ -42,8 +45,26 @@ public void shouldNotBeAbleToUseValueBeforeHardened() throws Exception
             value.get();
             fail( "Should fail" );
         }
-        catch ( TransientFailureException e )
-        {   // THEN
+        catch ( Exception e )
+        {
+            // THEN
+            assertThat( e, instanceOf( TransientDatabaseFailureException.class ) );
+        }
+    }
+
+    @Test
+    public void throwsWhenDelegateIsNotSet()
+    {
+        DelegateInvocationHandler<Value> handler = newDelegateInvocationHandler();
+
+        try
+        {
+            handler.invoke( null, Value.class.getDeclaredMethod( "get" ), new Object[0] );
```

Comment:
I know that the implementation doesn't use the proxy instance for anything, but passing null here still seems very odd.